gtsam/geometry/Point3.cpp

```diff
@@ -29,7 +29,10 @@ bool Point3::equals(const Point3 &q, double tol) const {
 }
 
 void Point3::print(const string& s) const {
-  cout << s << *this << endl;
+  if (s.size() != 0) {
+    cout << s << " ";
```

Comment:
Changes behavior, should revert. People might be relying on old behavior.

Reply:
Hmm I figured the printing would not affect actual behavior, but I guess for logging/metrics.